### DISCUSSION TITLE
What is the difference between pipe-viewer and youtube-viewer?

### DISCUSSION THREAD

**Discussion by Lokawn** [trizen/pipe-viewer#73] [General]
As the subject states I came across both these projects in `aur`, both of them have same description and same developer, So I just wanted a clarification as to what their individual purpose are apart from watching youtube videos.

Sorry if I missed something in the readme and this is just redundant.

**Comment by trizen** [comment]
The main difference is that `youtube-viewer` uses the YouTube API (and therefore requires an YouTube API key), while `pipe-viewer` parses the YouTube website directly (or can use the Invidious API to get the YouTube results), and does not require an YouTube API key.

There are some other minor differences, where `youtube-viewer` provides a few more features, and also allows log-in into your YouTube account, using OAuth2.0 (allowing to like/dislike videos, add videos to favorites, etc). The extra features being provided by the YouTube API.

On the other hand, in `pipe-viewer`, things like liking and disliking of videos, is done locally.

But, overall, both programs do the same thing (allowing to search for YouTube videos/channels/playlist and play the videos in a player of your choice).
